datafusion/src/execution/memory_manager.rs

```diff
@@ -340,6 +340,9 @@ impl MemoryManager {
             } else if current < min_per_rqt {
                 // if we cannot acquire at lease 1/2n memory, just wait for others
                 // to spill instead spill self frequently with limited total mem
+                debug!(
+                    "Cannot acquire minimum amount of memory {}, waiting for others to spill ...",
+                    human_readable_size(min_per_rqt));
```

Comment:
I'm also thinking if it makes sense to measure the time spent on the wait and log it?

Reply:
Yeah, that makes sense

Reply:
How do you like to also log memory manager status as well? 
``` rust
impl Display for MemoryManager {
    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
        write!(f,
               "MemoryManager usage statistics: total {}, trackers used {}, total {} requesters used: {}",
```

Reply:
sounds good

Reply:
One more thought, based on your measure timing comments above, probably we could `warn!` if we've waited for a longer time than some threshold? In case reproducing the race would be hard. `debug!` at the first round might not be revealed in the first place?

Reply:
`warn!` sounds better. Do you think we need to have `warn!` for all log here? Or just for longer time case?

Reply:
For the longer time case, I suggest.